ExecHashMap.hpp

```diff
@@ -0,0 +1,89 @@
+#include <iostream>
+#include <iterator>
+#include <numeric>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include <nlohmann/json.hpp>
+
+typedef struct IOPair {
```

Comment:
`IOPairs`

Reply:
Why do we need IOPair again?

Reply:
we need to declare an `IOPair` object to be thread_local in reporter, so it need to be a class or type, I thing just using `std::pair` also works in this case.